### PR TITLE
pios_usb_board_data: factor common

### PR DESCRIPTION
**Pull request by mlyle**

### CODE REVIEW

**Comment by mlyle** [comment]
jenkins, build artifacts please

File: make/firmware-defs.mk [hunk 1]
```diff
@@ -1,3 +1,19 @@
+ifneq ($(BOARD_INFO_DIR)x,x)
+include $(BOARD_INFO_DIR)/board-info.mk
+
+USB_VID_EXPAND := $(shell echo -n "$(USB_VID)" | sed -re "s/(.)/'\1',/g" -e 's/,$$//')
```
Comment (tracernz) [inline make/firmware-defs.mk]: Should call these USB_VENDOR and USB_PRODUCT instead IMO. VID and PID are the 16-bit integers.

Comment (mlyle) [reply make/firmware-defs.mk]: Yes, of course you're right-- will fixup tomorrow or Mon

Comment (tracernz) [reply make/firmware-defs.mk]: No prob. This looks good :+1:

**Comment by mlyle** [comment]
jenkins, build artifacts please

**Comment by mlyle** [comment]
jenkins, build artifacts please

**Comment by mlyle** [comment]
jenkins, build artifacts please

**Comment by mlyle** [comment]
jenkins, build artifacts please

**Comment by tracernz** [comment]
Looks good. :+1: macOS didn't like it though for some reason, and even gave a super helpful error message :/:
```
 CC        [bl|aq32]  flight/targets/bl/common/main.c
 CC        [bl|brai]  flight/targets/bl/common/main.c
arm-none-eabi-gcc: error: 12: No such file or directory
arm-none-eabi-gcc: error: 12: No such file or directory
arm-none-eabi-gcc: error: 12: No such file or directory
arm-none-eabi-gcc: error: 8: No such file or directory
make[1]: *** [/Users/Shared/Jenkins/Library/Developer/org.jenkins-ci.slave.jnlp/workspace/dronin/Nodes/osx/build/bl_brain/main.o] Error 1make[1]: 
*** [/Users/Shared/Jenkins/Library/Developer/org.jenkins-ci.slave.jnlp/workspace/dronin/Nodes/osx/build/bl_aq32/main.o] Error 1
make: *** [bl_aq32_bin] Error 2
make: *** Waiting for unfinished jobs....
make: *** [bl_brain_bin] Error 2
```

**Comment by mlyle** [comment]
ignoring macos right now because that at least i might be able to understand ;P

jenkins, build artifacts please

**Comment by mlyle** [comment]
jenkins, build artifacts please

**Comment by mlyle** [comment]
jenkins, build artifacts please

**Comment by mlyle** [comment]
jenkins, build artifacts please

**Comment by mlyle** [comment]
jenkins, build artifacts please

**Comment by mlyle** [comment]
i know what i'm doing and i'm not just pounding on a keyboard until it all works.  honest..  maybe.

jenkins, build artifacts please

**Comment by mlyle** [comment]
jenkins, build artifacts please

**Comment by mlyle** [comment]
jenkins, build artifacts please

**Comment by mlyle** [comment]
jenkins, build artifacts please

**Comment by mlyle** [comment]
master messed up because of disappearing merge tree ref

jenkins, build artifacts please

**Comment by mlyle** [comment]
Bench tested lux-- works fine, firmware size is the same between Linux & Windows & OSX and appears nominally identical (symbols/code in different order).  String table is sane.

Bench tested Sparky2.  It's OK when including #1672.  

Bench tested Sparky.

**Review by tracernz — APPROVED** [review]
It's a shame about the intermediate lib step but I guess not much option as long as we have to support Windows. :( Will merge when rebased. :)

**Comment by mlyle** [comment]
Rebased

**Comment by tracernz** [comment]
Damn, not quite.
```
make[1]: *** No rule to make target `board.c', needed by `/var/lib/jenkins/jobs/dron/workspace/Nodes/master/build/fw_pipxtreme/board.o'.  Stop.
```

**Comment by mlyle** [comment]
Should be good now..

**Pull request merged**